--- /sdcard/Download/llvm_android/base_builders.py	2020-11-14 20:16:20.122021028 +0800
+++ llvm_android/base_builders.py	2020-11-28 14:33:51.519630206 +0800
@@ -124,7 +124,8 @@
         ldflags = []
         # When cross compiling, toolchain libs won't work on target arch.
         if not self._is_cross_compiling():
-            ldflags.append(f'-L{self.toolchain.lib_dir}')
+            #ldflags.append(f'-L{self.toolchain.lib_dir}')
+            None
         return ldflags
 
     @property
@@ -144,7 +145,7 @@
     """Builder for autoconf targets."""
     src_dir: Path
     remove_install_dir: bool = True
-
+    
     @property
     def output_dir(self) -> Path:
         """The path for intermediate results."""
@@ -181,6 +182,11 @@
         cxxflags = super().cxxflags
         cxxflags.append('-stdlib=libc++')
         return cxxflags
+        
+    @property
+    def ldflags(self) -> List[str]:
+        ldflags = super().ldflags
+        return ldflags
 
     @property
     def config_flags(self) -> List[str]:
@@ -531,7 +537,9 @@
 
         defines['LLVM_ENABLE_PROJECTS'] = ';'.join(sorted(self.llvm_projects))
 
-        defines['LLVM_TARGETS_TO_BUILD'] = ';'.join(sorted(self.llvm_targets))
+        #defines['LLVM_TARGETS_TO_BUILD'] = ';'.join(sorted(self.llvm_targets))
+        defines['LLVM_TARGETS_TO_BUILD'] = 'all'
+        
         defines['LLVM_BUILD_LLVM_DYLIB'] = 'ON'
 
         if self.build_tags:
