--- a/llvm_android/base_builders.py	2021-09-15 06:00:45.315860391 +0800
+++ b/llvm_android/base_builders.py	2021-09-15 09:56:04.525460747 +0800
@@ -596,8 +596,14 @@
 
         defines['LLVM_ENABLE_PROJECTS'] = ';'.join(sorted(self.llvm_projects))
 
-        defines['LLVM_TARGETS_TO_BUILD'] = ';'.join(sorted(self.llvm_targets))
+        defines['LLVM_TARGETS_TO_BUILD']= 'all'
+        defines['LLVM_EXPERIMENTAL_TARGETS_TO_BUILD'] = 'AVR;RISCV'
         defines['LLVM_BUILD_LLVM_DYLIB'] = 'ON'
+        defines['LLVM_ENABLE_PIC'] = 'ON'
+        defines['CMAKE_BUILD_WITH_INSTALL_RPATH'] = 'ON'
+        defines['CMAKE_POLICY_DEFAULT_CMP0116'] = 'NEW'
+        defines['LLVM_TARGET_ARCH'] = 'AArch64'
+        defines['LLVM_HOST_TRIPLE'] = 'aarch64-unknown-linux-android'
 
         if self.build_tags:
             tags_str = ''.join(tag + ', ' for tag in self.build_tags)
