--- /sdcard/Download/llvm_android/hosts.py	2020-11-14 20:16:18.518687535 +0800
+++ llvm_android/hosts.py	2020-11-23 13:51:17.647505463 +0800
@@ -39,13 +39,19 @@
     def is_linux(self) -> bool:
         """Returns True if the given host is Linux."""
         return self == Host.Linux
-
+    
+    @property
+    def is_android(self) -> bool:
+        """Returns True if the given host is Linux."""
+        return self == Host.Android
+    
     @property
     def os_tag(self) -> str:
         """Returns the os tag of current Host."""
         return {
             Host.Darwin: 'darwin-x86',
             Host.Linux: 'linux-x86',
+            Host.Android: 'linux-aarch64',
             Host.Windows: 'windows-x86',
         }[self]
 
